File: leetsync/3227-find-missing-and-repeated-values/find-missing-and-repeated-values.java
class Solution {
    public int[] findMissingAndRepeatedValues(int[][] grid) {
        int n = grid.length;
        long N = (long) n * n;
        long gridSum = 0;
        long gridSqSum = 0;
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                gridSum += grid[i][j];
                gridSqSum += grid[i][j] * grid[i][j];
            }
        }
        long sum = (N * (N + 1)) / 2;
        long sqSum = (N * (N + 1) * (2 * N + 1)) / 6;

        long sumDiff = gridSum - sum;
        long sqDiff = gridSqSum - sqSum;

        int a = (int) ((sqDiff / sumDiff + sumDiff) / 2);
        int b = (int) ((sqDiff / sumDiff - sumDiff) / 2);
        return new int[] { a, b };
    }
}

File: leetsync/3227-find-missing-and-repeated-values/find-missing-and-repeated-values.go
func findMissingAndRepeatedValues(grid [][]int) []int {
	n := len(grid)
	N := n * n
	gridSum := 0
	gridSqSum := 0
	for i := 0; i < n; i++ {
		for j := 0; j < n; j++ {
			gridSum += grid[i][j]
			gridSqSum += grid[i][j] * grid[i][j]
		}
	}
	sum := (N * (N + 1)) / 2
	sqSum := (N * (N + 1) * (2*N + 1)) / 6

	sumDiff := gridSum - sum
	sqDiff := gridSqSum - sqSum

	a := (sqDiff/sumDiff + sumDiff) / 2
	b := (sqDiff/sumDiff - sumDiff) / 2
	return []int{a, b}
}